File: app/src/main/java/org/ametro/ui/navigation/entities/NavigationSubHeader.java
package org.ametro.ui.navigation.entities;

import androidx.annotation.Nullable;

public class NavigationSubHeader extends NavigationItem implements INavigationItemGroup {

    private final CharSequence text;
    private final NavigationItem[] items;

    public NavigationSubHeader(CharSequence text, NavigationItem[] items){
        this(text, items, null);
    }

    public NavigationSubHeader(CharSequence text, NavigationItem[] items, String tag){
        super(INVALID_ACTION);
        this.text = text;
        this.items = items;
        this.tag = tag;
    }

    public CharSequence getText() {
        return text;
    }

    public NavigationItem[] getItems() {
        return items;
    }
}

File: app/src/main/java/org/ametro/ui/navigation/entities/NavigationSubHeader.kt
package org.ametro.ui.navigation.entities

import org.ametro.ui.navigation.adapter.HolderFactory
import org.ametro.ui.navigation.adapter.NavigationSubHeaderHolderFactory

class NavigationSubHeader @JvmOverloads constructor(
    val text: CharSequence,
    override val items: Array<NavigationItem>,
    tag: String? = null
) : NavigationItem(INVALID_ACTION), NavigationItemGroup {
    override val factory = NavigationSubHeaderHolderFactory

    init {
        this.tag = tag
    }
}